include/seastar/testing/perf_tests.hh

```diff
@@ -167,42 +167,47 @@ class time_measurement {
     perf_stats _start_stats;
     perf_stats _total_stats;
 
+    linux_perf_event* _instructions_retired_counter = nullptr;
+
 public:
     [[gnu::always_inline]] [[gnu::hot]]
     void start_run(linux_perf_event* instructions_retired_counter = nullptr) {
+        _instructions_retired_counter = instructions_retired_counter;
         _total_time = { };
         auto t = clock_type::now();
         _run_start_time = t;
         _start_time = t;
-        _start_stats = perf_stats::snapshot(instructions_retired_counter);
+        _start_stats = perf_stats::snapshot(_instructions_retired_counter);
     }
 
     [[gnu::always_inline]] [[gnu::hot]]
-    performance_test::run_result stop_run(linux_perf_event* instructions_retired_counter = nullptr) {
+    performance_test::run_result stop_run() {
         auto t = clock_type::now();
         performance_test::run_result ret;
         if (_start_time == _run_start_time) {
             ret.duration = t - _start_time;
-            auto stats = perf_stats::snapshot(instructions_retired_counter);
+            auto stats = perf_stats::snapshot(_instructions_retired_counter);
             ret.stats = stats - _start_stats;
         } else {
             ret.duration = _total_time;
             ret.stats = _total_stats;
         }
+        _instructions_retired_counter = nullptr;
         return ret;
     }
 
     [[gnu::always_inline]] [[gnu::hot]]
-    void start_iteration(linux_perf_event* instructions_retired_counter = nullptr) {
+    void start_iteration() {
```

Comment:
This is a public API, does it break compatibility? Or is it not exposed?

Reply:
Thanks for your comment @avikivity. Yes, it could break compatibility: if someone was enterprising enough to build their own instruction counter class and pass it in this would break at compile time (or even if they just passed `nullptr`). This method is documented in perf-tests.md though always with 0 args.

So a non-breaking change would be to leave the signature of this method the same (with the pointer defaulting to nullptr has before), but when the pointer is null to use the saved instruction pointer. This would be compile-time compatible though the behavior changes: instructions are now correctly reported (arguably this fixes a bug, so I hope that is OK).

Reply:
I don't see start_iteration() in scylladb.git, probably the most aggressive Seastar user. Maybe we can let it slip.

Reply:
@avikivity - your call, I'm happy to change it or leave it.

Reply:
I haven't heard anything, how about we take the "let it slip" approach @avikivity mentioned?

Reply:
ping @avikivity.
Is it ok with you to merge this PR as is or are there any changes required?